vital/types/mesh.cxx

```diff
@@ -0,0 +1,628 @@
+  }
+  return *this;
+}
+
+
+/// Merge the data from another mesh into this one
+void
+mesh
+::merge(const mesh& other)
+{
+  const unsigned num_v = this->num_verts();
+  const unsigned num_e = this->num_edges();
+  faces_ = merge_face_arrays(*this->faces_,*other.faces_,verts_->size());
+  verts_->append(*other.verts_);
+
+  if (this->has_tex_coords() == TEX_COORD_NONE)
+  {
+    std::vector<vector_2d> tex;
+    if (other.has_tex_coords() == TEX_COORD_ON_VERT) {
+      tex = std::vector<vector_2d>(num_v, vector_2d(0,0));
+    }
+    else if (other.has_tex_coords() == TEX_COORD_ON_CORNER) {
+      tex = std::vector<vector_2d>(2*num_e, vector_2d(0,0));
+    }
+
+    tex.insert(tex.end(), other.tex_coords().begin(), other.tex_coords().end());
+    this->set_tex_coords(tex);
+  }
+  else if (this->has_tex_coords() == other.has_tex_coords())
+  {
+    this->tex_coords_.insert(this->tex_coords_.end(),
+                             other.tex_coords().begin(),
+                             other.tex_coords().end());
+  }
+
+  if (this->has_half_edges() && other.has_half_edges())
+    this->build_edge_graph();
+  else
+    this->half_edges_.clear();
+}
+
+
+/// Set the texture coordinates
+void
+mesh
+::set_tex_coords(const std::vector<vector_2d>& tc)
+{
+  if (tc.size() == this->num_verts())
+    tex_coord_status_ = TEX_COORD_ON_VERT;
+  else if (tc.size() == 2*this->num_edges())
+    tex_coord_status_ = TEX_COORD_ON_CORNER;
+  else
+    tex_coord_status_ = TEX_COORD_NONE;
+
+  tex_coords_ = tc;
+}
+
+
+/// Construct the half edges graph structure
+void
+mesh
+::build_edge_graph()
+{
+  const mesh_face_array_base& faces = this->faces();
+  std::vector<std::vector<unsigned int> > face_list(faces.size());
+  for (unsigned int f=0; f<faces.size(); ++f) {
+    face_list[f].resize(faces.num_verts(f));
+    for (unsigned int v=0; v<faces.num_verts(f); ++v)
+      face_list[f][v] =  faces(f,v);
+  }
+
+  half_edges_.build_from_ifs(face_list);
+}
+
+
+/// Compute vertex normals
+void
+mesh
+::compute_vertex_normals()
+{
+  if (!this->has_half_edges())
+    this->build_edge_graph();
+
+  const mesh_half_edge_set& half_edges = this->half_edges();
+  mesh_vertex_array<3>& verts = this->vertices<3>();
+
+  std::vector<vector_3d> normals(this->num_verts(), vector_3d(0,0,0));
+
+  for (unsigned int he=0; he < half_edges.size(); ++he) {
+    mesh_half_edge_set::f_const_iterator fi(he,half_edges);
+    if (fi->is_boundary())
+      continue;
+    unsigned int vp = fi->vert_index();
+    unsigned int v = (++fi)->vert_index();
+    unsigned int vn = (++fi)->vert_index();
+    normals[v] += mesh_tri_normal(verts[v],verts[vn],verts[vp]).normalized();
+  }
+
+  for (unsigned v=0; v<verts.size(); ++v)
+  {
+    normals[v].normalize();
+  }
+
+  verts.set_normals(normals);
+}
+
+
+/// Compute vertex normals using face normals
+void
+mesh
+::compute_vertex_normals_from_faces()
+{
+  if (!this->has_half_edges())
+    this->build_edge_graph();
```

Comment:
Braces around single line block.
I'll stop commenting on this at this point, but there are still some single statement blocks that need braces further down.